### ISSUE TITLE
Component: Accordion with buttons

### ISSUE THREAD

**Issue by Ahmed0ssama** [primefaces/primeng#17046]
### Describe the bug

I use accordion component from primeng v18, and I use buttons inside the accordion header but when I trigger clicking the button it automatically triggers opening or closing the accordion, I think this issue existed before in older versions, even when I faced this behavior in primeng v17 I used event.stopPropagation() to work properly.
The button works only if I clicked on it and held for a while then release the click, it's working only in this case.

### Environment

Angular 18 app

### Angular version

18.2.13

### PrimeNG version

18.0.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

22.11.0

### Expected behavior

I expect the button to work properly without opening or closing the accordion if I clicked in button area.

**Comment by Bock92** [comment]
Hi @Ahmed0ssama I don't believe this issue is related to `p-accordion`.

I tested the following code usign the `master` version of PrimeNG:
```
<p-accordion value="0">
    <p-accordion-panel value="0">
        <p-accordion-header>
            Header I
            <p-button label="ClickMe" (click)="testClick($event)"/>
        </p-accordion-header>
        <p-accordion-content>
            <p class="m-0">
                Header Context I
            </p>
        </p-accordion-content>
    </p-accordion-panel>
</p-accordion>
```

In my test, I added a `p-button` inside the `p-accordion-header` with the following handler:
```
class Foo {
  testClick = ($event: MouseEvent) => {
     $event.preventDefault();
     $event.stopPropagation();
     console.log('CLICKED');
     // Additional logic here...
  }
}
```

When I clicked the button, I only saw the 'CLICKED' message in the console. Conversely, when I clicked on the accordion header itself, it correctly triggered the open/close action.

If your issue differs from what I described, could you please share a reproducible example of the broken behavior? This will help us better understand the problem.

CC: @mertsincan

**Comment by Shusty-6th** [comment]
I found the cause of the bug. The reason is ripple!

See this stackblitz: https://stackblitz.com/edit/rqanxnr6?file=src%2Fmain.ts

The accordion with the button works badly, but when you disable ripple in main.ts, you can click the button without a problem.
I wonder how to get around this bug, to leave ripple enabled globally for the entire application, but disable it only for this one button.

**Comment by Ahmed0ssama** [comment]
> I found the cause of the bug. The reason is ripple!
> 
> See this stackblitz: https://stackblitz.com/edit/rqanxnr6?file=src%2Fmain.ts
> 
> The accordion with the button works badly, but when you disable ripple in main.ts, you can click the button without a problem. I wonder how to get around this bug, to leave ripple enabled globally for the entire application, but disable it only for this one button.

Yes, you are right when I tried to disable it globally buttons inside accordion header worked well, but I didn't find a way to enable it globally and only disabling it for this button (or this accordion header). Because I needed the ripple effect on the other pages.

**Comment by Ahmed0ssama** [comment]
> Hi [@Ahmed0ssama](https://github.com/Ahmed0ssama) I don't believe this issue is related to `p-accordion`.
> 
> I tested the following code usign the `master` version of PrimeNG:
> 
> ```
> <p-accordion value="0">
>     <p-accordion-panel value="0">
>         <p-accordion-header>
>             Header I
>             <p-button label="ClickMe" (click)="testClick($event)"/>
>         </p-accordion-header>
>         <p-accordion-content>
>             <p class="m-0">
>                 Header Context I
>             </p>
>         </p-accordion-content>
>     </p-accordion-panel>
> </p-accordion>
> ```
> 
> In my test, I added a `p-button` inside the `p-accordion-header` with the following handler:
> 
> ```
> class Foo {
>   testClick = ($event: MouseEvent) => {
>      $event.preventDefault();
>      $event.stopPropagation();
>      console.log('CLICKED');
>      // Additional logic here...
>   }
> }
> ```
> 
> When I clicked the button, I only saw the 'CLICKED' message in the console. Conversely, when I clicked on the accordion header itself, it correctly triggered the open/close action.
> 
> If your issue differs from what I described, could you please share a reproducible example of the broken behavior? This will help us better understand the problem.
> 
> CC: [@mertsincan](https://github.com/mertsincan)

When you click on the button it will log the message 'CLICKED' but at the same time it will trigger opening/closing the accordion if you were enabling "ripple" to the app, so it's a problem related to both (accordion behavior with ripple effect).

**Comment by JenoDK** [comment]
Update for anyone landing on this page: calling stopPropagation on `(mousedown)` seemed to work for me!
My code:
```
	<ng-template #headerTemplate let-title="title" let-model="model" let-accordionIndex="accordionIndex">
		<div class="mr-2 flex flex-wrap items-center gap-2">
			<span class="text-primary font-bold whitespace-nowrap">
				{{ title }}
			</span>
		<input
			pInputText
			type="text"
			class="grow w-4/12 p-1"
			(mousedown)="$event.stopPropagation()"
			(click)="$event.stopPropagation()"
			[(ngModel)]="model"
			(ngModelChange)="openAccordionTab(accordionIndex)"
		/>
		</div>
	</ng-template>
```
and later in the code 
```
<p-accordion-header>
    <ng-container
    	*ngTemplateOutlet="headerTemplate; context: {
    		title: translate('practle.scores.filter.groups'),
    		model: groupTextFilter,
    		accordionIndex: 1
    	}"
    >
    </ng-container>
</p-accordion-header>
```
